s2spy/time.py

```diff
@@ -70,12 +70,13 @@
 class AdventCalendar(BaseCalendar):
     """Countdown time to anticipated anchor date or period of interest."""
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
```

Comment:
I think I'd prefer a setter.

Reply:
Hm. How about keeping `max_lag` and `allow_overlap` together then?
For example:
```py
calendar = MonthlyCalendar(anchor='Dec', freq='1M')
calendar.set_max_lag(20, allow_overlap=True)
```

Reply:
I like that very much

Reply:
Also sounds good to me, to put them together.